### DISCUSSION TITLE
Setting state of a SpatieMediaLibraryFileUpload via Filament\Forms\Set

### DISCUSSION THREAD

**Discussion by Quadrubo** [filamentphp/filament#9194] [Ideas]
Hey,
currently it seems to me as if it is not possible, to set the state of the `SpatieMediaLibraryFileUpload ` using the ´Closure´ `Filament\Forms\Set`.

This syntax that works for a normal `FileUpload` doesn't work for the `SpatieMediaLibraryFileUpload`.
```php
$set('images', ['http://localhost/image.png']);
```

This is because, to account for the images being loaded from the Media model, the `getUploadedFileUsing` function has been modified to return `null` for every component without an attached record. I want to utilize the set functionality on the Create page so when a record doesn't exist yet.

`Filament\Forms\Components\SpatieMediaLibraryFileUpload:82`
```php
if (! $component->getRecord()) {
        return null;
    }
```

Would you be willing to accept a PR to change this behaviour so the `SpatieMediaLibraryFileUpload` would also accept 'uuid' passed as the `$file` argument instead? The `Media` model would then be looked up based on that. This would allow for the Closure to be used to set default files from the file system on the create page, etc.

```php
if (! $component->getRecord()) {
    $media = Media::where('uuid', $file)->first();
} else {
    /** @var ?Media $media */
    $media = $component->getRecord()->getRelationValue('media')->firstWhere('uuid', $file);
}
```
Usage:

```php
$set('images', ['989c9787-6361-466b-82a8-968da0e92cc2']);
```

If not, can someone give me alternative ways to solve this? When trying to pass the whole `Media` object as state, it seems that the function isn't even reached.

**Comment by torgodly** [comment]
Hey .. im trying to set default value from images i got in my disk .. can you help?
